hazelcast/src/main/java/com/hazelcast/nio/serialization/ClassDefinitionBuilder.java

```diff
@@ -183,19 +184,92 @@ public ClassDefinitionBuilder addFloatField(String fieldName) {
+     *
+     * @param fieldName name of the field that will be added to this class definition
+     * @return itself for chaining
+     * @throws HazelcastSerializationException if a field with same name already exists or
+     *                                         if this method is called after {@link ClassDefinitionBuilder#build()}
+     */
+    public ClassDefinitionBuilder addDecimalField(@Nonnull String fieldName) {
+        check(fieldName);
+        fieldDefinitions.add(new FieldDefinitionImpl(index++, fieldName, FieldType.DECIMAL, version));
+        return this;
+    }
+
+    /**
+     * Adds a time field consisting of hour, minute, seconds and nanos parts to the class definition
+     *
+     * @param fieldName name of the field that will be added to this class definition
+     * @return itself for chaining
+     * @throws HazelcastSerializationException if a field with same name already exists or
+     *                                         if this method is called after {@link ClassDefinitionBuilder#build()}
+     */
+    public ClassDefinitionBuilder addTimeField(@Nonnull String fieldName) {
+        check(fieldName);
+        fieldDefinitions.add(new FieldDefinitionImpl(index++, fieldName, FieldType.TIME, version));
+        return this;
+    }
+
+    /**
+     * Adds a date field consisting of year , month of the year and day of the month to the class definition
+     *
+     * @param fieldName name of the field that will be added to this class definition
+     * @return itself for chaining
+     * @throws HazelcastSerializationException if a field with same name already exists or
+     *                                         if this method is called after {@link ClassDefinitionBuilder#build()}
+     */
+    public ClassDefinitionBuilder addDateField(@Nonnull String fieldName) {
+        check(fieldName);
+        fieldDefinitions.add(new FieldDefinitionImpl(index++, fieldName, FieldType.DATE, version));
+        return this;
+    }
+
+    /**
+     * Adds a timestamp field consisting of
+     * year , month of the year and day of the month, hour, minute, seconds, nanos parts to the class definition
+     *
+     * @param fieldName name of the field that will be added to this class definition
+     * @return itself for chaining
+     * @throws HazelcastSerializationException if a field with same name already exists or
+     *                                         if this method is called after {@link ClassDefinitionBuilder#build()}
+     */
+    public ClassDefinitionBuilder addTimestampField(@Nonnull String fieldName) {
+        check(fieldName);
+        fieldDefinitions.add(new FieldDefinitionImpl(index++, fieldName, FieldType.TIMESTAMP, version));
+        return this;
+    }
+
+    /**
+     * Adds a timestamp with timezone field consisting of
+     * year , month of the year and day of the month, offset seconds , hour, minute, seconds, nanos parts
```

Comment:
`year , month` -> `year, month`

`month of the year and day of the month` -> `month of the year, day of the month`

Reply:
Same applies to `PortableReader`, `PortableWriter` and `GenericRecord` javadocs